src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java

```diff
@@ -148,6 +152,32 @@ public Mono<ResponseEntity<RequestAuthorizationResponseDto>> requestTransactionA
     ) {
         return requestAuthorizationRequestDto
                 .doOnNext(t -> log.info("RequestTransactionAuthorization for transactionId: [{}]", transactionId))
+                .map(updateAuthorizationRequest -> {
```

Comment:
maybe this lock is too restrictive, if we have  a db connection fails, we will block these calls, without the beneficial effects of retry

Reply:
We have to distinguish between DB connections fails, if they happen reading transaction from DB or writing event to event store.

Said that we could move this locking logic right before performing gateway api call, what do you think?

Reply:
Yes, in exception handler, but for a DB exception how can you tell were it happens? Fetching information from db or writing? 

IMHO moving locking just before performing api calls to gateway could be a right trade-off: for any error that could happen after communication have been initiated with gateway a retry should be avoid (there could be also network errors receiving response, such as timeout, but the authorization process could already be processed gateway side)